pkg/render/manager.go

```diff
@@ -87,26 +88,19 @@ func Manager(
 	clusterDomain string,
 	esLicenseType ElasticsearchLicenseType,
 ) (Component, error) {
-	tlsSecrets := []*corev1.Secret{}
-
-	if tlsKeyPair == nil {
-		var err error
-		svcDNSNames := dns.GetServiceDNSNames(ManagerServiceName, ManagerNamespace, clusterDomain)
-		svcDNSNames = append(svcDNSNames, "localhost")
-		tlsKeyPair, err = CreateOperatorTLSSecret(nil,
-			ManagerTLSSecretName,
-			ManagerSecretKeyName,
-			ManagerSecretCertName,
-			825*24*time.Hour, // 825days*24hours: Create cert with a max expiration that macOS 10.15 will accept
-			nil,
-			svcDNSNames...,
-		)
-		if err != nil {
-			return nil, err
-		}
-		tlsSecrets = []*corev1.Secret{tlsKeyPair}
+	ctx := context.Background()
+	svcDNSNames := dns.GetServiceDNSNames(ManagerServiceName, ManagerNamespace, clusterDomain)
+	svcDNSNames = append(svcDNSNames, "localhost")
+	certDur := 825 * 24 * time.Hour // 825days*24hours: Create cert with a max expiration that macOS 10.15 will accept
+
+	tlsKeyPair, err := EnsureCertificateSecret(
```

Comment:
To clarify my concern about `EnsureCertificateSecret`, the tlsKeyPair can be configured by the user if they want to provide certificates for the Manager UI manually. If they've specified this certificate without the correct dnsNames then this function will replace the user provided secret without the user knowing.

Reply:
We chatted about this over Slack. I've made the changes to check the ownerreferences of the secret. If the secret is invalid and the secret is owned by the component CR, we recreate the cert. If the secret is invalid and the secret is supplied by the user, we return an error and the controller is degraded.